src/java.desktop/unix/classes/sun/awt/wl/WLFrameDecoration.java

```diff
@@ -145,7 +154,10 @@ private int getButtonSpaceWidth() {
     }
 
     public void paint(final Graphics g) {
-        if (isUndecorated) return;
+        if (doesNotDrawClientSideDecoration()) {
+            needRepaint = false;
```

Comment:
Why this has become necessary?

Reply:
The flag is set to true when the server switches from CSD to SSD.

Reply:
>  The flag is set to true when the server switches from CSD to SSD.

Yes. But why setting the flag is necessary? Why not just exit with `return`?

Reply:
If the flag stays set, then this function might be called again unnecessarily.

Reply:
`needRepaint` has a very specific function and I'd like to keep it that way and not use it also for optimizations; unless `needRepaint = false;` is necessary for your code to work, please drop this line.

Reply:
Should be done now.